Frequency/Board/Map/Map.py

```diff
@@ -18,14 +18,16 @@ def __init__(self, resolution, tiles=None):
 
 
     def GenerateTiles(self):
-        maxTilesX = self.Resolution.X // 35
-        maxTilesY = self.Resolution.Y // 35
+        maxTiles = Vector2(18, 18)
+        maxTileSize = Vector2(self.Resolution.X // maxTiles.X, self.Resolution.Y // maxTiles.Y)
         tiles = []
 
-        for X in range(0, maxTilesX):
-            for Y in range(0, maxTilesY):
+        for X in range(0, maxTiles.X):
+            row = []
+            for Y in range(0, maxTiles.Y):
                 TileType = self.DetermineTileType(X, Y)
-                tiles.append(TileType(Vector2(X, Y)))
+                row.append(TileType(Vector2(X, Y), Vector2(maxTileSize.X, maxTileSize.Y)))
```

Comment:
maxTileSize is already a Vector2. So why creating an exactly copy as argument to create a new Tile?